src/core/search/autocompleteresponsetransformer.js

```diff
@@ -52,10 +52,17 @@ export default class AutoCompleteResponseTransformer {
   }
 
   static _transformAutoCompleteResult (result) {
+    let transformedFilter = {};
     if (result.filter) {
-      result.filter = this._transformFilter(result.filter);
```

Comment:
Attempting to reassign `result.filter` results in a read only error. The fix here is to pass data into the constructor for `AutoCompleteResult` using a new object that contains the transformed filter. Open to suggestions if there is a more graceful way of mitigating the issue.